lib/galaxy/tools/error_reports/plugins/gitlab.py

```
@@ -0,0 +1,144 @@
+        """Submit the error report to GitLab
+        """
+        log.info(self.gitlab)
+
+        if self.gitlab:
+            # Import the necessary libraries
+            import sys
+            import requests
+            if sys.version_info[0] < 3:
+                import urllib as urllib
+                import urlparse as urlparse
+            else:
+                import urllib.parse as urllib
+                urlparse = urllib
+
+            log.info(job.tool_id)
+            log.info(job.tool_version)
+            log.info(tool.tool_shed)
+
+            # Determine the ToolShed
+            ts_url_request = requests.get('http://' + str(tool.tool_shed))
+            ts_url = ts_url_request.url
+            log.info("Determined ToolShed is " + ts_url)
+
+            # Find the repo inside the ToolShed
+            ts_repo_request = requests.get(ts_url + "/api/repositories?tool_ids=" + str(job.tool_id))
+            ts_repo_request_data = ts_repo_request.json()
+            ts_repourl = None
+
+            for changeset, repoinfo in ts_repo_request_data.items():
+                if isinstance(repoinfo, dict):
+                    if 'repository' in repoinfo.keys():
+                        if 'remote_repository_url' in repoinfo['repository'].keys():
+                            ts_repourl = repoinfo['repository']['remote_repository_url']
+
+            log.info("Determine ToolShed Repository URL: " + ts_repourl)
+
+            if ts_repourl:
+                gitlab_projecturl = urlparse.urlparse(ts_repourl).path[1:]
+            else:
+                gitlab_projecturl = "/".join([self.gitlab_default_repo_owner,
+                                              self.gitlab_default_repo_name])
+
+            gitlab_urlencodedpath = urllib.quote_plus(gitlab_projecturl)
+
+            # Make sure we are always logged in, then retrieve the GitLab project
+            self.gitlab.auth()
+            gl_project = self.gitlab.projects.get(gitlab_urlencodedpath)
```

If the repository URL is not valid (i.e. same base as `gitlab_base_url`), will this step fail causing this bug reporter to fail (gracefully)?

---

It should throw an error then. It won't fail gracefully. That's still to be done.